crates/trie/sparse/src/trie.rs

```diff
@@ -820,6 +820,17 @@ impl<P> RevealedSparseTrie<P> {
                                 hash.filter(|_| path.is_empty()),
                             );
                             updates.updated_nodes.insert(path.clone(), branch_node);
+                        } else if self
+                            .branch_node_tree_masks
+                            .get(&path)
+                            .is_some_and(|mask| !mask.is_empty()) ||
+                            self.branch_node_hash_masks
+                                .get(&path)
+                                .is_some_and(|mask| !mask.is_empty())
+                        {
+                            // If new tree and hash masks are empty, but previously they weren't, we
+                            // need to remove the node.
+                            updates.removed_nodes.insert(path.clone());
                         }
```

Comment:
we could just always delete them, but this might be more precise

Reply:
yes, but then it creates a bunch of no-op deletions, so this is significantly better